mqtt/build/linux/compliance.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+###############################################################################
+# This script runs eclipse compliance tests against the broker
+###############################################################################
+
+set -e
+
+###############################################################################
+# Define Environment Variables
+###############################################################################
+# Get directory of running script
+DIR=$(cd "$(dirname "$0")" && pwd)
+
+BUILD_REPOSITORY_LOCALPATH=${BUILD_REPOSITORY_LOCALPATH:-$DIR/../../..}
+PROJECT_ROOT=${BUILD_REPOSITORY_LOCALPATH}/mqtt
+SCRIPT_NAME=$(basename "$0")
+CARGO="${CARGO_HOME:-"$HOME/.cargo"}/bin/cargo"
+RELEASE="true"
+
+###############################################################################
+# Download tests and make certificate
+###############################################################################
+
+$DIR/certgen.sh
+git clone https://github.com/lfitchett/paho.mqtt.testing.git
+
+###############################################################################
+# Start broker and run tests
+###############################################################################
+set +e
```

Comment:
Why `-e` here https://github.com/Azure/iotedge/pull/2802/files#diff-0f2e075fd2852867d4ab80bf4a1b4ea6R7
but `+e` in this line?

Reply:
for the above lines, I want an error to kill the script. Below, I manually grab the error code of the tests `result=$?` because I want to kill the broker even if the tests fail.